gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp

```diff
@@ -0,0 +1,116 @@
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace std::chrono_literals;
+
+/// Tests the gazebo_ros_wheelslip plugin
+class GazeboRosWheelSlipTest : public gazebo::ServerFixture
+{
+};
+
+TEST_F(GazeboRosWheelSlipTest, SetSlipCompliance)
+{
+  // Load test world and start paused
+  this->Load("worlds/gazebo_ros_wheelslip.world", true);
+
+  // World
+  auto world = gazebo::physics::get_world();
+  ASSERT_NE(nullptr, world);
+
+  // Create node
+  auto node = std::make_shared<rclcpp::Node>("test_gazebo_ros_wheelslip");
+  ASSERT_NE(nullptr, node);
+
+  auto parameters_client_rear = std::make_shared<rclcpp::SyncParametersClient>(
+    node,
+    "trisphere_cycle_slip0/wheel_slip_rear");
+
+  std::map<std::string, double> parameters_rear_pairs = {
+    {"slip_compliance_unitless_lateral", 5},
+    {"slip_compliance_unitless_lateral/wheel_rear_left", 0.0},
+    {"slip_compliance_unitless_lateral/wheel_rear_right", 5},
+    {"slip_compliance_unitless_longitudinal", 6},
+    {"slip_compliance_unitless_longitudinal/wheel_rear_left", 4},
+    {"slip_compliance_unitless_longitudinal/wheel_rear_right", 6}};
+
+  int counter = 0;
+  while (!parameters_client_rear->wait_for_service(1s)) {
+    if (!rclcpp::ok()) {
+      rclcpp::shutdown();
+    }
+    std::cout << "service not available, waiting again..." << std::endl;
+    counter++;
+    if (counter > 5) {FAIL();}
+  }
+
+  std::vector<std::string> parameters_rear_names;
+  for (auto & element : parameters_rear_pairs) {
+    parameters_rear_names.push_back(element.first);
+  }
+
+  // Verify parameters were set as per the SDF, negative values should be replaced by 0
+  auto parameters_rear = parameters_client_rear->get_parameters(parameters_rear_names);
+  for (auto & parameter : parameters_rear) {
+    ASSERT_EQ(parameters_rear_pairs[parameter.get_name()], parameter.as_double());
+  }
+
+  // Set slip compliance for one wheel, verify others remain unchanged
+  parameters_rear_pairs["slip_compliance_unitless_lateral/wheel_rear_left"] = 3.0;
+  std::vector<rclcpp::Parameter> change_param = {
+    rclcpp::Parameter("slip_compliance_unitless_lateral/wheel_rear_left", 3.0)};
+  auto result = parameters_client_rear->set_parameters(change_param);
+  ASSERT_TRUE(result[0].successful);
+
+  parameters_rear = parameters_client_rear->get_parameters(parameters_rear_names);
+  for (auto & parameter : parameters_rear) {
+    ASSERT_EQ(parameters_rear_pairs[parameter.get_name()], parameter.as_double());
+  }
+
+  // Set global slip compliance parameters, which should override the ones for wheels
+  std::vector<rclcpp::Parameter> change_params_global = {
+    rclcpp::Parameter("slip_compliance_unitless_lateral", 10.0),
+    rclcpp::Parameter("slip_compliance_unitless_longitudinal", 11.0)
+  };
+  result = parameters_client_rear->set_parameters(change_params_global);
+  ASSERT_TRUE(result[0].successful);
+  ASSERT_TRUE(result[1].successful);
+
+  std::this_thread::sleep_for(2s);
```

Comment:
I presume this sleep is to give the plugin time to update its parameters since we're setting individual wheel slip parameters _after_ successfully setting the global parameters. It's unfortunate that we have to do this, but I don't see a way around it. It would be good to add a comment here explaining why we need to sleep.